XiEditor/SidebarViewController.swift

```diff
@@ -0,0 +1,142 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Cocoa
+
+class SidebarViewController: NSViewController {
+
+    @IBOutlet weak var sidebar: NSOutlineView!
+
+    var theme: Theme {
+        return (NSApplication.shared.delegate as! AppDelegate).theme
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Setup notification observer for when the sidebar items change
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(self.updateItems),
+            name: NSNotification.Name(rawValue: "sidebarItemsChanged"),
+            object: nil
+        )
+
+        self.updateItems()
+    }
+
+    override func viewWillAppear() {
+        sidebar.backgroundColor = self.theme.background
+    }
+
+    public func themeChanged(_ theme: String) {
+        sidebar.backgroundColor = self.theme.background
+    }
+
+    /// TODO: Sync the structure (expanded/collapsed/selected) of each windows sidebar
+    public func syncStructure() {
+        
+    }
+
+    @objc private func updateItems() {
+        self.sidebar.reloadData()
+    }
+
+}
+
+// MARK: - Sidebar NSOutlineViewDataSource
+
+extension SidebarViewController: NSOutlineViewDataSource {
+
+    // Number of items in the sidebar
+    func outlineView(_ outlineView: NSOutlineView, numberOfChildrenOfItem item: Any?) -> Int {
+        guard let item = item as? FileSystemItem else { return sidebarItems.items.count }
+        return item.getNumberOfChildren()
+    }
+
+    // Items to be added to sidebar
+    func outlineView(_ outlineView: NSOutlineView, child index: Int, ofItem item: Any?) -> Any {
+        guard let item = item as? FileSystemItem else { return sidebarItems.items[index] }
+        return item.getChild(at: index)
+    }
+
+    // Whether rows are expandable by an arrow
+    func outlineView(_ outlineView: NSOutlineView, isItemExpandable item: Any) -> Bool {
+        guard let item = item as? FileSystemItem else { return false }
+        return item.getNumberOfChildren() != 0
+    }
+
+    // Height of each row
+    func outlineView(_ outlineView: NSOutlineView, heightOfRowByItem item: Any) -> CGFloat {
+        return 20.0
+    }
+
+    // When a row is clicked on should it be selected
+    func outlineView(_ outlineView: NSOutlineView, shouldSelectItem item: Any) -> Bool {
+        guard let item = item as? FileSystemItem else { return false }
+        return !item.isDirectory
```

Comment:
Even if it's a directory, it should still be clickable, and I also suggest that double clicking expands the directory, similar to the arrow click. I found clicking on arrows to expand directories to be jarring and inaccurate (especially for folders that have a lot of subfolders, much like our own xi-core repo), since the arrow isn't very big by design and it takes too much effort and accuracy to just expand directories in the sidebar. 

Double-clicking to expand folders has become quintessential UX (for good reason!) at this point, and users familiar with desktop environments expect it. You can see many other apps that make use of a directory sidebar (like other editors, or Finder, or XCode) allow users to double-click on directory rows to expand them.